src/main/kotlin/io/github/gradlenexus/publishplugin/InitializeNexusStagingRepository.kt

```diff
@@ -88,9 +87,13 @@ constructor(
     }
 
     private fun replacePublishingRepoUrl(url: URI) {
-        val publishing = project.the<PublishingExtension>()
-        val repository = publishing.repositories.getByName(repository.get().name) as MavenArtifactRepository
-        logger.info("Updating URL of publishing repository '{}' to '{}'", repository.name, url)
-        repository.setUrl(url.toString())
+        project.allprojects {
+            val publishing = extensions.findByType<PublishingExtension>()
+            if (publishing != null) {
+                val repository = publishing.repositories.getByName(repository.get().name) as MavenArtifactRepository
+                logger.info("Updating URL of publishing repository '{}' to '{}'", repository.name, url)
+                repository.setUrl(url.toString())
```

Comment:
Is this safe concurrency-wise?

Reply:
Since the field is not volatile, you're probably right that this isn't 100% safe. I guess we could make sure it's set correctly using a `doFirst {}` action on every `PublishToMavenRepository` task (which is guaranteed to be executed in the same thread as the task's main action) and loading the staging repo URL from a thread-safe location (e.g. a `BuildService` with a `ConcurrentHashMap<String, URI>` for Gradle 6.1+). Other ideas?

Reply:
replacePublishingRepoUrl is called from InitializeNexusStagingRepository TaskAction which is executed in the root project.

However, it seems to update objects related to other projects which might even execute at the same time.

---

Have you tried `repository.setUrl(Provider<String>)`? Does it work?

Reply:
I completely missed that `setUrl` takes `Object`. That should work! 👍

Reply:
It did work. Thanks for the idea! 🙂